csharp/tools/MauiModelTester/MainPage.xaml.cs

```diff
@@ -0,0 +1,158 @@
+                                                               if (task.IsFaulted)
+                                                               {
+                                                                   DisplayAlert("Error", task.Exception.Message, "OK");
+                                                               }
+                                                           });
+                                                   });
+    }
+
+    private void OnRunClicked(object sender, EventArgs e)
+    {
+        RunAsync().ContinueWith(resultTask =>
+                                {
+                                    if (resultTask.IsFaulted)
+                                    {
+                                        MainThread.BeginInvokeOnMainThread(
+                                            () => DisplayAlert("Error", resultTask.Exception.Message, "OK"));
+                                    }
+                                });
+    }
+
+    private async Task RunAsync()
+    {
+        await ClearResult();
+
+        int iterations = Iterations.Text == string.Empty ? 10 : int.Parse(Iterations.Text);
+
+        await SetBusy(true);
+
+        // create inference session if it doesn't exist or EP has changed
+        await CreateInferenceSession();
+
+        await Task.Run(() => _inferenceSession.Run(iterations));
+
+        await SetBusy(false);
+
+        ShowResults();
+    }
+
+    private async Task SetBusy(bool busy)
+    {
+        await MainThread.InvokeOnMainThreadAsync(() =>
+                                                 {
+                                                     BusyIndicator.IsRunning = busy;
+                                                     BusyIndicator.IsVisible = busy;
+                                                     // disable controls that would create a new session or another
+                                                     // Run call until we're done with the current Run.
+                                                     ExecutionProviderOptions.IsEnabled = !busy;
+                                                     RunButton.IsEnabled = !busy;
+                                                 });
+    }
+
+    private async Task ClearResult()
+    {
+        await MainThread.InvokeOnMainThreadAsync(() =>
+                                                 { TestResults.Clear(); });
+    }
+
+    private void ShowResults()
+    {
+        var createResults = () =>
+        {
+            var stats = _inferenceSession.PerfStats;
+            var label = new Label { TextColor = Colors.GhostWhite };
+
+            label.Text = $"Model load time: {stats.LoadTime.TotalMilliseconds:F4} ms\n";
+            label.Text += $"Warmup run time: {stats.WarmupTime.TotalMilliseconds:F4} ms\n\n";
+            label.Text += string.Join('\n', stats.GetRunStatsReport(true));
```

Comment:
would it be useful to log the perf stats too? may be easier to copy-paste from log output